src/ParallelAlgorithms/Interpolation/Actions/InterpolatorReceivePoints.hpp

```diff
@@ -80,12 +89,34 @@ struct ReceivePoints {
                                          Metavariables>>(*vars_holders)
                   .infos;
 
-          // Add the target interpolation points at this temporal_id.
-          vars_infos.emplace(std::make_pair(
-              temporal_id,
-              intrp::Vars::Info<VolumeDim, typename InterpolationTargetTag::
-                                               vars_to_interpolate_to_target>{
-                  std::move(block_logical_coords)}));
+          // Add the new target interpolation points at this temporal_id. There
+          // are two conditions that allow us to overwrite the current target
+          // points. Either
+          //
+          //  1. There are no current target points at the temporal_id, OR
+          //  2. There are target points already at this temporal_id, but the
+          //     iteration of the new target points is greater than the
+          //     iteration of the current target points.
+          //
+          // If we already have target points and the iteration of the new
+          // points is less than or equal to the iteration of the current target
+          // points, then we ignore the new points. The new points are outdated
+          // and we definitely didn't have any of the new target points in our
+          // element by the fact that we have already received the next
+          // iteration of points.
+          //
+          // Whenever we overwrite the target points, we also empty the
+          // `interpolation_is_done_for_these_elements` (by virtue of a default
+          // constructed `intrp::Vars::Info`) so that we always check every
+          // element for this new set of target points.
+          if (vars_infos.count(temporal_id) == 0 or
+              vars_infos.at(temporal_id).iteration < iteration) {
```

Comment:
Make it error if the new iteration is the same as the old iteration. This is a bug.

Reply:
@wthrowe Pushed a fixup for this

Reply:
Looks good.